File: src/main/kotlin/net/yttravelers/spotshare/entities/RouteDetailRestaurant.kt
package net.yttravelers.spotshare.entities

import com.fasterxml.jackson.annotation.JsonIgnore
import java.time.LocalDateTime
import javax.persistence.Table
import javax.persistence.*

@Entity
@Table(name = "routeDetailRestaurant")
class RouteDetailRestaurant {
    @Id
    var routeDetailId: Int? = null

    @JsonIgnore
    @OneToOne
    @JoinColumn(name = "routeDetailId")
    @MapsId
    lateinit var routeDetail: RouteDetail

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "restaurantId")
    lateinit var restaurant: Restaurant

    var restaurantRate: Int? = null

    var restaurantMinutes: Int? = null

    //TODO: 区分の設計
//    var restaurantMealKindDiv:

    var restaurantStartTime: LocalDateTime? = null

    var restaurantEndTime: LocalDateTime? = null

    var restaurantCost: Int? = null
}

File: resources/js/src/app/shared/model/route-detail-restaurant.ts
import { IRestaurant } from './restaurant';

/**
 * ルート詳細飲食店
 */
export interface IRouteDetailRestaurant {
  /** ルート詳細ID */
  routeDetailId?: number;
  /** 飲食店ID */
  restaurantId?: number;
  /** 行きたい度 */
  restaurantRate?: number;
  /** 飲食時間 */
  restaurantMinutes?: number;
  /** 食事種類区分 */
  restaurantMealKindDiv?: number;
  /** 食事種類区分名称 */
  restaurantMealKindDivName?: string;
  /** 飲食費用 */
  restaurantCost?: number;
  /** 飲食店 */
  restaurant?: IRestaurant;
}
